### PR TITLE
UTBot multi-IDE support

### PR DESCRIPTION
**Pull request by tamarinvs19**
### IDEs
Supported plugin building for PyCharm (Community/Professional) and GoLand.
image

By default plugin will be support languages according to `ideType` (parameter in `gradle.properties`):

| ideType | UTBot language support | IDE |
|-----------|------------|---|
| IC | Java, Python | IDEA Community |
| IU | Java, Python, JavaScript, Go | IDEA Ultimate |
| PC | Python | PyCharm Community |
| PY | Python, JavaScript | PyCharm Professional |
| GO | Go | GoLand |

For example, if you set `ideType=PC` (in `gradle.properties`) to build utbot plugin for PyCharm Community.

<details>
  <summary>Language to IDEs</summary>
 
| Language | ideTypes |
|-----------|------------|
| Java | IC, IU |
| Python | IC, IU, PC, PY |
| JavaScript | IU, PY |
| Go | IU, GO |
  
</details>

__Additional functionality:__ you can set `projectType=Community` to build plugin without Go and JavaScript for IDEA Ultimate.

__NB:__ Plugin for IU works in IC (only for Java and Python), for IC works in IU; plugin for PY works in PC and for PC works in PY.

### Settings

UTBot settings have been separated to "common" and <language>-specific (now only java-specific). Java-specific settings have been moved to UnitTestBot sub-page.

In PyCharm or GoLand build Java settings will not be included.

Also `Experimental language support` settings now is enabled by default.

### Manual tests

Build plugin (`$ gradle buildPlugin`) with different `ideType` and check that UTBot supports all suitable languages.

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [ ] The functionality I've repaired, changed or added is covered with **automated tests**.
- [x] **Manual tests** have been provided optionally.
- [ ] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-intellij-js/build.gradle.kts [hunk 1]
```diff
@@ -46,10 +50,16 @@ intellij {
 
     val androidPlugins = listOf("org.jetbrains.android")
 
-    val jvmPlugins = listOf(
+    val jvmPlugins = mutableListOf(
         "java"
     )
 
+    val kotlinPlugins = mutableListOf(
+        "org.jetbrains.kotlin"
```
Comment (Vassiliy-Kudryashov) [inline utbot-intellij-js/build.gradle.kts:58]: Could be immutable either

File: utbot-intellij-main/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+intellij {
+
+    val androidPlugins = listOf("org.jetbrains.android")
+
+    val jvmPlugins = mutableListOf(
+        "java"
+    )
+
+    val kotlinPlugins = mutableListOf(
+        "org.jetbrains.kotlin"
```
Comment (Vassiliy-Kudryashov) [inline utbot-intellij-main/build.gradle.kts:38]: And one more mutable list with no reason.

File: utbot-intellij-python/build.gradle.kts [hunk 3]
```diff
@@ -46,10 +50,16 @@ intellij {
 
     val androidPlugins = listOf("org.jetbrains.android")
 
-    val jvmPlugins = listOf(
+    val jvmPlugins = mutableListOf(
         "java"
     )
 
+    val kotlinPlugins = mutableListOf(
+        "org.jetbrains.kotlin"
```
Comment (Vassiliy-Kudryashov) [inline utbot-intellij-python/build.gradle.kts:58]: Maybe listOf() will work too.

File: utbot-intellij-go/build.gradle.kts [hunk 4]
```diff
@@ -46,10 +50,16 @@ intellij {
 
     val androidPlugins = listOf("org.jetbrains.android")
 
-    val jvmPlugins = listOf(
+    val jvmPlugins = mutableListOf(
         "java"
     )
 
+    val kotlinPlugins = mutableListOf(
```
[the remark this answers is not in the document]
Comment (tamarinvs19) [reply utbot-intellij-go/build.gradle.kts]: No, replaced it by `listOf(...)`

File: utbot-intellij-js/src/main/kotlin/org/utbot/intellij/plugin/js/JsDialogProcessor.kt [hunk 5]
```diff
@@ -201,10 +203,11 @@ object JsDialogProcessor {
                             testDir.add(temp)
                             testDir.findFile(testFileName)!!
                         }
-                        val testFileEditor = CodeInsightUtil.positionCursor(project, testPsiFile, testPsiFile) as Editor
-                        unblockDocument(project, testFileEditor.document)
-                        testFileEditor.document.setText(generatedCode)
-                        unblockDocument(project, testFileEditor.document)
+//                        val testFileEditor = CodeInsightUtil.positionCursor(project, testPsiFile, testPsiFile) as Editor
```
Comment (Vassiliy-Kudryashov) [inline utbot-intellij-js/src/main/kotlin/org/utbot/intellij/plugin/js/JsDialogProcessor.kt]: Please remove all newly commented code

**Review by Vassiliy-Kudryashov — APPROVED** [review]

File: gradle.properties [hunk 6]
```diff
@@ -11,14 +12,17 @@ ultimateEdition=Ultimate
 # IU, IC, PC, PY
 # IC for AndroidStudio
 ideType=IC
-ideVersion=232.8660.185
+ideaVersion=232.8660.185
+pycharmVersion=2023.2
```
Comment (EgorkaKulikov) [inline gradle.properties:16]: Where id `golandVersion`?

File: gradle.properties [hunk 7]
```diff
@@ -11,14 +12,17 @@ ultimateEdition=Ultimate
...
 buildType=NOJS
 
+javaIde=IC,IU
 pythonIde=IC,IU,PC,PY
-jsIde=IU,PY,WS
+jsIde=IU,PY
 jsBuild=ALL
-goIde=IU
+goIde=IU,GO
+androidStudioIde=IC
```
Comment (EgorkaKulikov) [inline gradle.properties:25]: Why not IU?

**Review by EgorkaKulikov — APPROVED** [review]

File: gradle.properties (hunk 6 above)
Comment (tamarinvs19) [reply gradle.properties:16]: Fixed

File: gradle.properties (hunk 7 above)
Comment (tamarinvs19) [reply gradle.properties]: Removed

**Pull request merged**